### ISSUE TITLE
Unhelpful error message when stay unquoted brace in double quoted string.

### ISSUE THREAD

**Issue by cognominal** [rakudo/rakudo#4123]
## The Problem

  >say "{";  my $i;
  Variable '$i' is not declared

Forgetting to quote an opening brace within double quoted string leads to a confusing message. The error message "variable $foo is not declared" does not help. The error message should say that the material following the brace is interpreted as block code.
Maybe many other unhelpful error messages may be caused by a stray brace with different stuff following it.

Note that the "my $i" can be far away from the brace making the problem less conspicuous.

Syntax highlighting may help to make obvious the user error.  Not in VScode.
But raku syntax highlighting is humongous can of worms that makes raku less than
palatable in most editing environments. Probably solved in comma.

**Comment by jnthn** [comment]
> Probably solved in comma.

Yes, it shows up in quoted text color, not as the keywords, which is a good clue.

![image](https://user-images.githubusercontent.com/50259/102651167-5ac40000-416c-11eb-8b0d-42c26d9f7ef8.png)

Probably we can keep track of it we're inside a interpolated closure inside of a quoted string (and of note, track the start position) using a dynamic variable, glance at that when we panic, and provide a hint along with the error.

**Comment by cognominal** [comment]
Got caught again.
This shortcoming hurts particularly bad when transliterating in raku Perl 5 code that prints code with a lot  of
braces. A give away is that many lines before the reported error don't need update to compile correctly  in raku. That's first feel good but this is bad because they are in fact interpreted as part of a double string instead of normal code.

A workaround would be to be able to set a default for double quotes that does not interpret braces.
